src/sinks/util/http.rs

```diff
@@ -21,11 +21,64 @@ pub struct HttpService {
 }
 
 impl HttpService {
+    pub fn builder() -> HttpServiceBuilder {
+        HttpServiceBuilder::new()
+    }
+
     pub fn new<F>(request_builder: F) -> Self
     where
         F: Fn(Vec<u8>) -> hyper::Request<Vec<u8>> + Sync + Send + 'static,
     {
-        let https = HttpsConnector::new(4).expect("TLS initialization failed");
+        Self::builder().build(request_builder)
+    }
+}
+
+/// A builder for `HttpService`s
+pub struct HttpServiceBuilder {
+    threads: usize,
+    verify_certificate: bool,
+}
+
+impl HttpServiceBuilder {
+    fn new() -> Self {
+        Self {
+            threads: 4,
+            verify_certificate: true,
+        }
+    }
+
+    pub fn build<F>(&self, request_builder: F) -> HttpService
+    where
+        F: Fn(Vec<u8>) -> hyper::Request<Vec<u8>> + Sync + Send + 'static,
+    {
+        HttpService::from((self, request_builder))
+    }
+
+    /// Set the number of threads used by the `HttpService`
+    pub fn threads(&mut self, threads: usize) -> &mut Self {
+        self.threads = threads;
+        self
+    }
+
+    /// Verify the remote server's certificate
+    pub fn verify_certificate(&mut self, verify: bool) -> &mut Self {
+        self.verify_certificate = verify;
+        self
+    }
+}
+
+impl<F> From<(&HttpServiceBuilder, F)> for HttpService
```

Comment:
I don't think this from impl makes much sense can we move this to the build fn in Builder?

Reply:
Done